### PR TITLE
[R1281] Google pay load payment data

### PR DESCRIPTION
**Pull request by mike-ridge**
- Add new method to RyftPaymentService to allow loading data from a payment session using the client secret and refactor existing listener behaviour to allow this
- Add amount & currency to the PaymentSessionResponse object
- Load payment session when google pay is clicked
- Implement loading of google pay payment data after payment session is loaded successfully

### CODE REVIEW

File: ryft-ui/src/main/java/com/ryftpay/android/ui/service/request/GooglePayRequestFactory.kt [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            )
+        }.toString()
+    )
+
+    private fun createMerchantInfo(
+        merchantInfo: MerchantInfo
+    ): JSONObject = JSONObject()
+        .put("merchantName", merchantInfo.displayName)
+
+    private fun createTokenizationSpecification(
```
Comment (ryft-rkirby) [inline ryft-ui/src/main/java/com/ryftpay/android/ui/service/request/GooglePayRequestFactory.kt]: Might be worth exposing this publicy should any client want to place the GooglePay button on a product page/outside of checkout and handle the GooglePay interaction themselves.

We could add this in the next iteration however, with an example in the demo app.

File: ryft-ui/src/main/java/com/ryftpay/android/ui/service/request/TransactionInfo.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package com.ryftpay.android.ui.service.request
+
+import com.ryftpay.android.core.model.payment.PaymentSession
+import java.util.Currency
+
+internal data class TransactionInfo(
```
Comment (ryft-rkirby) [inline ryft-ui/src/main/java/com/ryftpay/android/ui/service/request/TransactionInfo.kt]: is this GooglePay specific data?
Perhaps we place this under a googlepay package?

**Review by ryft-rkirby — APPROVED** [review]

File: ryft-ui/src/main/java/com/ryftpay/android/ui/service/request/GooglePayRequestFactory.kt (hunk 1 above)
Comment (mike-ridge) [reply ryft-ui/src/main/java/com/ryftpay/android/ui/service/request/GooglePayRequestFactory.kt]: yeah good shout, I'll pick this up separately 👍

File: ryft-ui/src/main/java/com/ryftpay/android/ui/service/request/TransactionInfo.kt (hunk 2 above)
Comment (mike-ridge) [reply ryft-ui/src/main/java/com/ryftpay/android/ui/service/request/TransactionInfo.kt]: yeah it is - updated 👍

**Pull request merged**